### PR TITLE
Use raster images with Android adaptive icons

### LINKED ISSUE #1104: [Bug] Resizetizer fails with IndexOutOfRange
### Description

After adding the Prism icon svg file to the Prism Demo an error is encountered with an IndexOutOfRangeException.

### Steps to Reproduce

1. Add Prism Icon
2. Build

[prismicon.svg.zip](https://github.com/dotnet/maui/files/6555232/prismicon.svg.zip)

### Expected Behavior

The build completes using the Prism logo. Alternatively a more useful error message.

### Actual Behavior

The build fails with the error message: 

```
C:\Users\dsieg\.nuget\packages\microsoft.maui.resizetizer\6.0.100-preview.4.634\buildTransitive\Microsoft.Maui.Resizetizer.targets(423,9): error MAUI0000: System.IndexOutOfRangeException: Index was outside the bounds of the array. [C:\repos\PrismLib\Prism.Maui\sample\PrismMauiDemo\PrismMauiDemo.csproj]
C:\Users\dsieg\.nuget\packages\microsoft.maui.resizetizer\6.0.100-preview.4.634\buildTransitive\Microsoft.Maui.Resizetizer.targets(423,9): error MAUI0000:    at Svg2VectorDrawable.Svg2Vector.ParseDimension(SvgTree avg, XmlNode nNode) in Svg2VectorDrawable.Net.dll:token 0x6000014+0x0 [C:\repos\PrismLib\Prism.Maui\sample\PrismMauiDemo\PrismMauiDemo.csproj]
C:\Users\dsieg\.nuget\packages\microsoft.maui.resizetizer\6.0.100-preview.4.634\buildTransitive\Microsoft.Maui.Resizetizer.targets(423,9): error MAUI0000:    at Svg2VectorDrawable.Svg2Vector.Parse(String file) in Svg2VectorDrawable.Net.dll:token 0x6000011+0x4f [C:\repos\PrismLib\Prism.Maui\sample\PrismMauiDemo\PrismMauiDemo.csproj]
C:\Users\dsieg\.nuget\packages\microsoft.maui.resizetizer\6.0.100-preview.4.634\buildTransitive\Microsoft.Maui.Resizetizer.targets(423,9): error MAUI0000:    at Svg2VectorDrawable.Svg2Vector.ParseSvgToXml(String inputSvgFilename, Stream outputStream) in Svg2VectorDrawable.Net.dll:token 0x6000021+0x0 [C:\repos\PrismLib\Prism.Maui\sample\PrismMauiDemo\PrismMauiDemo.csproj]
C:\Users\dsieg\.nuget\packages\microsoft.maui.resizetizer\6.0.100-preview.4.634\buildTransitive\Microsoft.Maui.Resizetizer.targets(423,9): error MAUI0000:    at Svg2VectorDrawable.Svg2Vector.Convert(String inputSvgFilename, String outputXmlFilename) in Svg2VectorDrawable.Net.dll:token 0x6000020+0x7 [C:\repos\PrismLib\Prism.Maui\sample\PrismMauiDemo\PrismMauiDemo.csproj]
C:\Users\dsieg\.nuget\packages\microsoft.maui.resizetizer\6.0.100-preview.4.634\buildTransitive\Microsoft.Maui.Resizetizer.targets(423,9): error MAUI0000:    at Microsoft.Maui.Resizetizer.AndroidAdaptiveIconGenerator.Generate() in Microsoft.Maui.Resizetizer.dll:token 0x600000e+0x132 [C:\repos\PrismLib\Prism.Maui\sample\PrismMauiDemo\PrismMauiDemo.csproj]
C:\Users\dsieg\.nuget\packages\microsoft.maui.resizetizer\6.0.100-preview.4.634\buildTransitive\Microsoft.Maui.Resizetizer.targets(423,9): error MAUI0000:    at Microsoft.Maui.Resizetizer.ResizetizeImages.ProcessAppIcon(ResizeImageInfo img, ConcurrentBag`1 resizedImages) in Microsoft.Maui.Resizetizer.dll:token 0x60000c8+0x51 [C:\repos\PrismLib\Prism.Maui\sample\PrismMauiDemo\PrismMauiDemo.csproj]
C:\Users\dsieg\.nuget\packages\microsoft.maui.resizetizer\6.0.100-preview.4.634\buildTransitive\Microsoft.Maui.Resizetizer.targets(423,9): error MAUI0000:    at Microsoft.Maui.Resizetizer.ResizetizeImages.<>c__DisplayClass26_0.<ExecuteAsync>b__0(ResizeImageInfo img) in Microsoft.Maui.Resizetizer.dll:token 0x60000f3+0x14 [C:\repos\PrismLib\Prism.Maui\sample\PrismMauiDemo\PrismMauiDemo.csproj]
C:\Users\dsieg\.nuget\packages\microsoft.maui.resizetizer\6.0.100-preview.4.634\buildTransitive\Microsoft.Maui.Resizetizer.targets(423,9): error MAUI0000:    at Microsoft.Maui.Resizetizer.AsyncTaskExtensions.<>c__DisplayClass0_0`1.<ParallelForEach>b__0(TSource s) in Microsoft.Maui.Resizetizer.dll:token 0x60000f1+0x0 [C:\repos\PrismLib\Prism.Maui\sample\PrismMauiDemo\PrismMauiDemo.csproj]
```

### Basic Information

- Version with issue: pre4
- Last known good version: n/a
- IDE: dotnet cli

### LINKED ISSUE #6455: .Net MAUI app crashing phone
_This issue has been moved from [a ticket on Developer Community](https://developercommunity.visualstudio.com/t/Net-MAUI-app-crashing-phone/1521554)._

---
[severity:I'm unable to use this version]
I wanted to try .Net MAUI with VS 2022 preview 3, and tried to run the initial application on my actual phone (Samsung Galaxy S7), the app started without any problem and the hot reload kinda worked.
But as soon as I ended my debug session (and came back on Home), my phone became very very laggy, and "Samsung Home Experience" kept crashing every 4-5 seconds, making my whole display fully reset.

I soon realized that if I was quick enough to reach Settings, the crash would stop for the time I left Home, and started to explore the settings to find a solution.

I decided to go in Applications, find the app that had been installed by Visual Studio and uninstall it, but as soon as the blade with the icon and the name of the app should have been displayed when scrolling through my applications, Settings crashed the same way as Samsung Home Experience did. 

Settings crashes as soon as it should display it.
Samsung Home Screens takes a few seconds, but when it initializes, it loads the icons 1 by 1, and the one from the MAUI app should be the very last one in my phone.
So I am pretty sure the "loading" of the icon makes anything crash on my phone.

The only solution I found (for anyone who'd face the same problem) was to go to applications in settings again, select them all, unselect those I want to keep (until I would reach the "danger zone" that would make my Settings crash) and uninstall all the rest (hence, having the MAUI app selected)
Since my phone is not super recent that would take a while, but once it has finished, my Home stops crashing, only problem remaining is that I lost any app whose name would have made it sorted after the MAUI app.

I made 2 more experiments in order to collect more data 
- I fully reseted developer mode on my phone before trying to do the process all over again, crash came back
- I installed a basic mobile app from Android Studio, no crash, so the problem really is between Visual Studio Preview 3 or .Net MAUI with my specific phone.

Hope that helps, do not hesitate to notify me if this is found and fixed, since I don't have an intel CPU, emulation is not really a solution for me.

---
#### Feedback Bot on 9/5/2021, 08:22 PM: 

<p>We have directed your feedback to the appropriate engineering team for further evaluation. The team will review the feedback and notify you about the next steps.</p>

#### Paul Wang [MSFT] on 9/6/2021, 03:35 AM: 

<p>Hi, thanks for your feedback. Do you have this problem on Windows or MAC? And what is the build number?</p>

#### Benoit Hoffman on 9/6/2021, 11:56 PM: 

<p>Hi, sorry for not specifying at least my OS.</p>
<p>My OS is Microsoft Windows 10 Education version 10.0.18362 Build 18362.<br>
The .net preview version I run (info from terminal) is 6.0.100-preview.7.21379.14</p>
<p>and here is the list of intalled products fetched from VS:<br>

```
Microsoft Visual Studio Enterprise 2022 Preview
Version 17.0.0 Preview 3.1
VisualStudio.17.Preview/17.0.0-pre.3.1+31612.314
Microsoft .NET Framework
Version 4.8.03752

Installed Version: Enterprise

Visual C++ 2022 00476-70000-00000-AA218
Microsoft Visual C++ 2022

.NET Core Debugging with WSL 1.0
.NET Core Debugging with WSL

ADL Tools Service Provider 1.0
This package contains services used by Data Lake tools

ASA Service Provider 1.0

ASP.NET and Web Tools 2019 17.0.414.6609
ASP.NET and Web Tools 2019

ASP.NET Web Frameworks and Tools 2019 17.0.414.6609

Azure App Service Tools v3.0.0 17.0.414.6609
Azure App Service Tools v3.0.0

Azure Data Lake Tools for Visual Studio 2.6.2000.0
Microsoft Azure Data Lake Tools for Visual Studio

Azure Functions and Web Jobs Tools 17.0.414.6609
Azure Functions and Web Jobs Tools

Azure Stream Analytics Tools for Visual Studio 2.6.2000.0
Microsoft Azure Stream Analytics Tools for Visual Studio

C# Tools 4.0.0-3.21403.23+d51d1a7338e32d3a3cd1e5452665964ee8ec15df
C# components used in the IDE. Depending on your project type and settings, a different version of the compiler may be used.

Common Azure Tools 1.10
Provides common services for use by Azure Mobile Services and Microsoft Azure Tools.

Extensibility Message Bus 1.2.6 (master@34d6af2)
Provides common messaging-based MEF services for loosely coupled Visual Studio extension components communication and integration.

Fabric.DiagnosticEvents 1.0
Fabric Diagnostic Events

Microsoft Azure Hive Query Language Service 2.6.2000.0
Language service for Hive query

Microsoft Azure Service Fabric Tools for Visual Studio 17.0
Microsoft Azure Service Fabric Tools for Visual Studio

Microsoft Azure Stream Analytics Language Service 2.6.2000.0
Language service for Azure Stream Analytics

Microsoft Azure Tools for Visual Studio 2.9
Support for Azure Cloud Services projects

Microsoft JVM Debugger 1.0
Provides support for connecting the Visual Studio debugger to JDWP compatible Java Virtual Machines

Microsoft Library Manager 2.1.134+45632ee938.RR
Install client-side libraries easily to any web project

Microsoft MI-Based Debugger 1.0
Provides support for connecting Visual Studio to MI compatible debuggers

Microsoft Visual C++ Wizards 1.0
Microsoft Visual C++ Wizards

Microsoft Visual Studio Tools for Containers 1.2
Develop, run, validate your ASP.NET Core applications in the target environment. F5 your application directly into a container with debugging, or CTRL + F5 to edit & refresh your app without having to rebuild the container.

Microsoft Visual Studio VC Package 1.0
Microsoft Visual Studio VC Package

Mono Debugging for Visual Studio 17.0.6 (1ebf2f7)
Support for debugging Mono processes with Visual Studio.

NuGet Package Manager 6.0.0
NuGet Package Manager in Visual Studio. For more information about NuGet, visit docs.nuget.org

ProjectServicesPackage Extension 1.0
ProjectServicesPackage Visual Studio Extension Detailed Info

Razor (ASP.NET Core) 17.0.0.2138002+9e71a625d64fa38b323fa6ac52154e6b1bde703d
Provides languages services for ASP.NET Core Razor.

Snapshot Debugging Extension 1.0
Snapshot Debugging Visual Studio Extension Detailed Info

SQL Server Data Tools 17.0.62108.04180
Microsoft SQL Server Data Tools

Test Adapter for Boost.Test 1.0
Enables Visual Studio’s testing tools with unit tests written for Boost.Test. The use terms and Third Party Notices are available in the extension installation directory.

Test Adapter for Google Test 1.0
Enables Visual Studio’s testing tools with unit tests written for Google Test. The use terms and Third Party Notices are available in the extension installation directory.

ToolWindowHostedEditor 1.0
Hosting json editor into a tool window

TypeScript Tools 17.0.0730.2001
TypeScript Tools for Microsoft Visual Studio

Visual Basic Tools 4.0.0-3.21403.23+d51d1a7338e32d3a3cd1e5452665964ee8ec15df
Visual Basic components used in the IDE. Depending on your project type and settings, a different version of the compiler may be used.

Visual F# Tools 17.0.0-beta.21370.1+9b96c1bde17e9af532ae34eacbfc12d1628183d7
Microsoft Visual F# Tools

Visual Studio Code Debug Adapter Host Package 1.0
Interop layer for hosting Visual Studio Code debug adapters in Visual Studio

Visual Studio Container Tools Extensions 1.0
View, manage, and diagnose containers within Visual Studio.

Visual Studio IntelliCode 2.2
AI-assisted development for Visual Studio.

Visual Studio Tools for CMake 1.0
Visual Studio Tools for CMake

Visual Studio Tools for Containers 1.0
Visual Studio Tools for Containers

VisualStudio.DeviceLog 1.0
Information about my package

VisualStudio.Foo 1.0
Information about my package

VisualStudio.Mac 1.0
Mac Extension for Visual Studio

Xamarin 17.0.0.145 (main@bdc8369)
Visual Studio extension to enable development for Xamarin.iOS and Xamarin.Android.

Xamarin Designer 17.0.0.58 (remotes/origin/d17-0@927d548a1)
Visual Studio extension to enable Xamarin Designer tools in Visual Studio.

Xamarin Templates 17.0.3 (a84fb28)
Templates for building iOS, Android, and Windows apps with Xamarin and Xamarin.Forms.

Xamarin.Android SDK 11.4.99.70 (main/7e1558c)
Xamarin.Android Reference Assemblies and MSBuild support.
Mono: c633fe9
Java.Interop: xamarin/java.interop@4fb7c14
ProGuard: Guardsquare/proguard@912d149
SQLite: xamarin/sqlite@a575761
Xamarin.Android Tools: xamarin/xamarin-android-tools@623332d

Xamarin.iOS and Xamarin.Mac SDK 14.99.0.294 (13b1b3c45)
Xamarin.iOS and Xamarin.Mac Reference Assemblies and MSBuild support.
```

<hr>
<p>If that helps, when creating my .net MAUI project, I could not find the template in VS although I had everything from doc installed, so I succesfully created it from command line <code>dotnet new maui -n _projectname_</code></p>

#### Feedback Bot on 9/7/2021, 07:08 PM: 

<p>This issue is currently being investigated. Our team will get back to you if either more information is needed, a workaround is available, or the issue is resolved.</p>

#### Feedback Bot on 9/8/2021, 11:36 AM: 

<p>We have directed your feedback to the appropriate engineering team for further evaluation. The team will review the feedback and notify you about the next steps.</p>

#### Jonathan Peppers [MSFT] on 9/8/2021, 02:05 PM: 

<p>Can you attach <a target='_blank' href="https://docs.microsoft.com/en-us/xamarin/android/deploy-test/debugging/android-debug-log">adb logcat output</a> when your phone is acting strange? There should be some error message or something showing what is happening. I’m not sure what could cause a problem like this.</p>
<p>Does a Release build of the app have any issues?</p>
<p><strong>Why do we ask for more info?</strong><br>
We try to reproduce all issues reported with the information provided in the description and comments. When we can’t reproduce the issue, we ask you for more information so we can resolve the issue as quickly and efficiently as possible.<br>
In our <a target='_blank' href="https://aka.ms/vsfeedbackguidelines/#writing-a-good-bug-report-or-feature-suggestion">guidelines</a>, you can get tips on how to provide clear and simple reproducible steps.</p>

#### Benoit Hoffman on 9/9/2021, 02:02 AM: 

<p>Hi Jonathan, and thanks for looking into that, I really want to try MAUI out 😦</p>
<p>Same problem seems to happen with Release build (so the app works fine, but phone dies)</p>
<p>You may find a 6000 lines log record from my phone in the attached Android_Logs.txt, most of those lines seems to be useless debug messages coming from the app itself (ConnectivityService Tag) and could easily be filtered out but I left them there just in case 😃</p>
<p>The operations I performed was:</p>
<ul>
<li>start the app from VS</li>
<li>go into settings to avoid repetitive crash</li>
<li>stop the debugging from VS</li>
<li>start recording of the logs</li>
<li>go to home and wait for 2-3 crashes to happen</li>
<li>stop recording</li>
</ul>
<p>Really hope it helps,</p>
<p>Cheers!</p>

#### Jonathan Peppers [MSFT] on 9/9/2021, 06:49 AM: 

(private comment, text removed)
#### Benoit Hoffman on 9/9/2021, 08:42 AM: 

(private comment, text removed)
#### Benoit Hoffman on 10/13/2021, 02:33 AM: 

<p>Hi <a target='_blank' href="https://developercommunity2.visualstudio.com/user/4907eedc-8e83-6418-970d-be28acc2b5b2/profile">Jonathan Peppers [MSFT]</a>,</p>
<p>I thought you might be interested to learn that I installed the new version of Visual Studio Preview 5, and the problem is still exactly the same for me 😦.</p>
<p>I can’t believe that it’s common, otherwise you guys would have already figured it out, but I am so sad I still cannot play with MAUI 😦.</p>
<p>Additional note: this test with preview 5 has been made on another computer than the first tests, so the fault is most probably between MAUI and the phone, not the dev environment.</p>
<p>Cheers!</p>

#### Jonathan Peppers [MSFT] on 10/13/2021, 07:06 AM: 

<p>Does the problem occur with a plain .NET 6 Android app? This is <code>dotnet new android</code> or the template that just says <code>Android Application</code>.</p>

#### Benoit Hoffman on 10/13/2021, 09:54 AM: 

<p>I get a weird “the path is not in a legal form” error when building from VS, but the app is nevertheless installed on the phone, works as expected, and no bug like the one described in my ticket is to be seen.</p>
<p>Just to be clear, the “plain Android app” I tried was created with the <code>dotnet new android</code> command, while the .Net MAUI app I tried in VS preview 4 and 5 were created using the MAUI app template from VS directly.</p>

#### Jonathan Peppers [MSFT] on 10/13/2021, 11:03 AM: 

<p>I think the “path is not in a legal form” error is something that happens when you open a <code>.csproj</code> file and didn’t save the <code>.sln</code> file in VS yet. We have an issue somewhere filed about this.</p>
<p>So your original problem only occurs with MAUI and nothing appears to be wrong with plain .NET 6 Android apps?</p>

#### Benoit Hoffman on 10/13/2021, 11:26 AM: 

<p>Exactly, the plain .NET 6 Android app works fine and my phone has 0 problem.</p>
<p>Btw, were you able to get your hands on my Android Logs this time?</p>

#### Benoit Hoffman on 10/13/2021, 11:50 AM: 

<p>Also, I was able to get my hands on another old phone just for trial (Samsung J5 2016), and the app works fine, no bug to be seen.</p>

#### Jonathan Peppers [MSFT] on 10/13/2021, 00:28 PM: 

<p>No, we’re not seeing those logs.</p>
<p>Can you try creating a <a target='_blank' href="https://gist.github.com">gist</a> (or some similar site), and just share the link to it? Thanks!</p>

#### Benoit Hoffman on 10/13/2021, 01:07 PM: 

<p>Sure thing <a target='_blank' href="https://developercommunity2.visualstudio.com/user/4907eedc-8e83-6418-970d-be28acc2b5b2/profile">Jonathan Peppers [MSFT]</a>!</p>
<p>There it is: <a target='_blank' href="https://gist.github.com/Petit-Pas/b64d2c1c92a44cab10174ad53131eebb">https://gist.github.com/Petit-Pas/b64d2c1c92a44cab10174ad53131eebb</a></p>

#### Jonathan Peppers [MSFT] on 10/13/2021, 02:54 PM: 

<p>This part might be interesting:</p>
<pre><code><span class="hljs-number">09</span><span class="hljs-number">-09</span> <span class="hljs-number">10</span>:<span class="hljs-number">49</span>:<span class="hljs-number">31.964</span>	Samsung SM-G935F	Debug	<span class="hljs-number">15252</span>	LauncherActivityInfo	packageName: com.companyname.ZzzMAUIProject, supportTheme: <span class="hljs-literal">false</span>, height: <span class="hljs-number">912</span>, width: <span class="hljs-number">912</span>, density: <span class="hljs-number">640</span>
<span class="hljs-number">09</span><span class="hljs-number">-09</span> <span class="hljs-number">10</span>:<span class="hljs-number">49</span>:<span class="hljs-number">31.964</span>	Samsung SM-G935F	Info	<span class="hljs-number">15252</span>	ApplicationPackageManager	<span class="hljs-keyword">return</span> adaptive icon <span class="hljs-keyword">for</span> <span class="hljs-literal">NULL</span>
<span class="hljs-number">09</span><span class="hljs-number">-09</span> <span class="hljs-number">10</span>:<span class="hljs-number">49</span>:<span class="hljs-number">31.964</span>	Samsung SM-G935F	Debug	<span class="hljs-number">4836</span>	PackageManager	getSelectedMetaData : packageName(com.companyname.ZzzMAUIProject) <span class="hljs-keyword">or</span> Metadata strings {[Ljava.lang.String;@<span class="hljs-number">59b</span>9037}
</code></pre>
<p>What if you delete the fancy Maui icon?</p>
<pre><code class="language-xml"><span class="hljs-tag">&lt;<span class="hljs-name">MauiIcon</span>
  <span class="hljs-attr">Include</span>=<span class="hljs-string">"Resources\appicon.svg"</span>
  <span class="hljs-attr">ForegroundFile</span>=<span class="hljs-string">"Resources\appiconfg.svg"</span>
  <span class="hljs-attr">Color</span>=<span class="hljs-string">"#512BD4"</span> /&gt;</span>
</code></pre>
<p>Then just copy over the icon from the <code>dotnet new android</code> app?</p>
<p>This line has the icon:</p>
<ul>
<li><a target='_blank' href="https://github.com/xamarin/xamarin-android/blob/fbbdc9f9a80b12fc70fab13f40b02844f75da1c5/src/Microsoft.Android.Templates/android/AndroidManifest.xml#L3">https://github.com/xamarin/xamarin-android/blob/fbbdc9f9a80b12fc70fab13f40b02844f75da1c5/src/Microsoft.Android.Templates/android/AndroidManifest.xml#L3</a></li>
</ul>
<p>And then the <code>mipmap</code> files:</p>
<ul>
<li><a target='_blank' href="https://github.com/xamarin/xamarin-android/tree/main/src/Microsoft.Android.Templates/android/Resources">https://github.com/xamarin/xamarin-android/tree/main/src/Microsoft.Android.Templates/android/Resources</a></li>
</ul>

#### Benoit Hoffman on 10/14/2021, 00:54 AM: 

<p>Ha! I always said it was probably about the icon being loaded 😄<br>
Here is what I did:</p>
<p>1st, I went to see the so called fancy Maui icon just to make sure we were on the same page, and it could be worth noting that the line referencing an icon in my csproj was &lt;MauiImage&gt; and not &lt;MauiIcon&gt;. So I first tried to just edit this, no change.</p>
<p>Then I looked at the mipmap files (forgive me for any misuse of terms, I am not very familiar with Android dev environment) and realized they were none in my MAUI project under Platforms/Android/Resources/, so I copy pasted every mipmap folder from the Android app to the Platforms/Android/Resources.<br>
Finally I changed the &lt;application&gt; in AndroidManifest (since I guess here is the place I was supposed to “copy over the icon from the <code>dotnet new android</code> app”)</p>
<p>For the record, here is the line that was generated by MAUI template (note that it uses mipmap ressources, even though they had not been generated in my project):<br>
<code>&lt;application android:allowBackup=&quot;true&quot; android:icon=&quot;@mipmap/appicon&quot; android:roundIcon=&quot;@mipmap/appicon_round&quot; android:supportsRtl=&quot;true&quot;&gt;&lt;/application&gt;</code><br>
And here is the line generated by <code>dotnet new android</code> (note that it does add android:label, and the ressources used are differents):<br>
<code>&lt;application android:allowBackup=&quot;true&quot; android:icon=&quot;@mipmap/ic_launcher&quot; android:label=&quot;@string/app_name&quot; android:roundIcon=&quot;@mipmap/ic_launcher_round&quot; android:supportsRtl=&quot;true&quot;&gt;&lt;/application&gt;</code></p>
<p>I then had trouble about the color ic_launcher_background not being defined, so I copy pasted that line only from plain Android to MAUI Platforms/Android/Resources/values/colors.xml.<br>
Same for string/app_name, I copied the file into values/ folder.</p>
<p>After that I was finally able to try to build the app aand <em>drum rolls</em> it works!<br>
My application works as before, but my home screen does not crash anymore, I even dared scrolling in my phone until I could see the icon, and even though it’s now showing xamarin instead of MAUI, no crash to report.</p>
<p>Now the remaining questions are:</p>
<ul>
<li>Was the generation of my project somehow incorrect? Not having mipmap files generated and yet having the &lt;application&gt; referencing them seems odd. Even though the app and icon were working fine on the other, older phone</li>
<li>If, for whatever build magic, it was correct, what triggered the problem with my phone ? (Android 8.0, API 26, I realized I never said it yet)</li>
</ul>
<p>Anyway, thanks for your help <a target='_blank' href="https://developercommunity2.visualstudio.com/user/4907eedc-8e83-6418-970d-be28acc2b5b2/profile">Jonathan Peppers [MSFT]</a>!<br>
I’ll keep refreshing this thread anyway, in case you need more insight from my side to find the source of the problem itself, since I believe you’ll probably want this fixed for the Release Candidate in (hopefully) Q1 2022 😃</p>

---
### Original Solutions
(no solutions)

### PR DESCRIPTION
**Pull request by mattleibow**
### Description of Change ###

The default implementation of resizetizer to generate Android adaptive icons was to try and generate the Android Vector version of the input SVG. The main issue is the fact that SVG is a massive spec so the Android Vector was a subset with reduce functionality. In addition to that, the converter for the SVG to Android Vector is very rudimentary.

As a result, only a minute set of vectors could be used as app icons. 

This PR goes with the work to convert the input SVG files into rasters, and then uses those rasters as the components of the adaptive icons. This allows the "adaptive" nature of the icons, but avoids the SVG conversion issues.

Fixes #1104
Fixes https://github.com/xamarin/xamarin-android/issues/6455

### Additions made ###

* Adds logic to generate raster versions of the input SVG app icons

### PR Checklist ###

- [ ] Targets the correct branch 
- [ ] Tests are passing (or failures are unrelated)
- [ ] Targets a single property for a single control (or intertwined few properties)
- [ ] Adds the property to the appropriate interface
- [ ] Avoids any changes not essential to the handler property
- [ ] Adds the mapping to the PropertyMapper in the handler
- [ ] Adds the mapping method to the WinUI, Android, iOS, and Standard aspects of the handler
- [ ] Implements the actual property updates (usually in extension methods in the Platform section of Core)
- [ ] Tags ported renderer methods with [PortHandler]
- [ ] Adds an example of the property to the sample project (MainPage)
- [ ] Adds the property to the stub class
- [ ] Implements basic property tests in DeviceTests

#### Does this PR touch anything that might affect accessibility?
- [ ] Does this PR introduce a new control? (If yes, add an example using SemanticProperties to the SemanticsPage)
- [ ] APIs that modify focusability?
- [ ] APIs that modify any text property on a control?
- [ ] Does this PR modify view nesting or view arrangement in anyway?
- [ ] Is there the smallest possibility that your PR will change accessibility? 
- [ ] I'm not sure, please help me

If any of the above checkboxes apply to your PR, then the PR will need to provide testing to demonstrate that accessibility still works.

### CODE REVIEW

File: src/SingleProject/Resizetizer/src/ResizetizeImages.cs [hunk 1]
```diff
@@ -11,6 +11,8 @@ namespace Microsoft.Maui.Resizetizer
 {
 	public class ResizetizeImages : MauiAsyncTask, ILogger
 	{
+		internal bool AllowVectorAdaptiveIcons = false;
```
Comment (mattleibow) [inline src/SingleProject/Resizetizer/src/ResizetizeImages.cs:14]: For testing purposes as well as to potentially bring back the vector support if there is demand.

File: src/SingleProject/Resizetizer/src/AndroidAdaptiveIconGenerator.cs [hunk 2]
```diff
@@ -42,57 +42,115 @@ public IEnumerable<ResizedImageInfo> Generate()
...
-			var foregroundIsVector = Info.ForegroundIsVector;
 			var foregroundExists = File.Exists(foregroundFile);
+			var foregroundIsVector = !foregroundExists || (UseVectors && Info.ForegroundIsVector);
+			var foregroundExt = foregroundIsVector ? ".xml" : ".png";
+			var foregroundDestFilename = AppIconName + "_foreground" + foregroundExt;
 
-			Logger.Log("Looking for Foreground File: " + foregroundFile);
-
-			// If we have vectors we can emit an adaptive icon
-			if (backgroundIsVector && (foregroundIsVector || !foregroundExists))
```
Comment (mattleibow) [inline src/SingleProject/Resizetizer/src/AndroidAdaptiveIconGenerator.cs:54]: Always generate an app icon - unlike previously where one was only generated if both images were vectors.

**Review by jonathanpeppers — COMMENTED** [review]
Can you share some examples of what the AndroidResources looks like?

One concern I have is a report that using MAUI app icons crashes the launcher on their device:

* https://developercommunity.visualstudio.com/t/Net-MAUI-app-crashing-phone/1521554
* https://github.com/xamarin/xamarin-android/issues/6455

(FYI I don't know why the issue was moved it to xamarin-android either, lol)

But I've not been able to reproduce this on my devices -- and their workaround was to copy the icon from the `dotnet new android` template, and not use MAUI app icons.

Do we think this might make this issue better/worse? Do you think it's worth making a sample they could test this out?

File: src/SingleProject/Resizetizer/src/AndroidAdaptiveIconGenerator.cs [hunk 3]
```diff
@@ -42,57 +42,115 @@ public IEnumerable<ResizedImageInfo> Generate()
...
-				if (!fileInfo.Directory.Exists)
-					fileInfo.Directory.Create();
-
 				Logger.Log("Converting Background SVG to Android Drawable Vector: " + backgroundFile);
-				Svg2VectorDrawable.Svg2Vector.Convert(backgroundFile, backgroundDestination);
 
-				var foregroundDestination = Path.Combine(fullIntermediateOutputPath.FullName, "drawable", AppIconName + "_foreground.xml");
-				fileInfo = new FileInfo(foregroundDestination);
+				var destination = Path.Combine(fullIntermediateOutputPath.FullName, "drawable-v24", backgroundDestFilename);
+				var fileInfo = new FileInfo(destination);
```
[the remark this answers is not in the document]
Comment (mattleibow) [reply src/SingleProject/Resizetizer/src/AndroidAdaptiveIconGenerator.cs:60]: I think that was always there. Let me change that.

File: src/SingleProject/Resizetizer/src/AndroidAdaptiveIconGenerator.cs [hunk 4]
```diff
@@ -42,57 +42,110 @@ public IEnumerable<ResizedImageInfo> Generate()
...
+			if (backgroundIsVector)
+			{
+				Logger.Log("Converting Background SVG to Android Drawable Vector: " + backgroundFile);
 
-			// If we have vectors we can emit an adaptive icon
-			if (backgroundIsVector && (foregroundIsVector || !foregroundExists))
+				var dir = Path.Combine(fullIntermediateOutputPath.FullName, "drawable-v24");
+				var destination = Path.Combine(dir, backgroundDestFilename);
+				if (!Directory.Exists(dir))
+					Directory.CreateDirectory(dir);
```
Comment (jonathanpeppers) [inline src/SingleProject/Resizetizer/src/AndroidAdaptiveIconGenerator.cs:62]: Ah another idea, you can just call `Directory.CreateDirectory(dir);` and if it doesn't exist it's a no-op.
```suggestion
				Directory.CreateDirectory(dir);
```

File: src/SingleProject/Resizetizer/src/AndroidAdaptiveIconGenerator.cs [hunk 5]
```diff
@@ -42,57 +42,110 @@ public IEnumerable<ResizedImageInfo> Generate()
...
+				Logger.Log("Converting Background SVG to PNG: " + backgroundFile);
 
-				Logger.Log("Converting Background SVG to Android Drawable Vector: " + backgroundFile);
-				Svg2VectorDrawable.Svg2Vector.Convert(backgroundFile, backgroundDestination);
+				foreach (var dpi in DpiPath.Android.AppIconParts)
+				{
+					var dir = Path.Combine(fullIntermediateOutputPath.FullName, dpi.Path);
+					var destination = Path.Combine(dir, backgroundDestFilename);
+					if (!Directory.Exists(dir))
+						Directory.CreateDirectory(dir);
```
Comment (jonathanpeppers) [inline src/SingleProject/Resizetizer/src/AndroidAdaptiveIconGenerator.cs:77]: Probably in here a couple places now.

File: src/SingleProject/Resizetizer/src/AndroidAdaptiveIconGenerator.cs (hunk 4 above)
Comment (mattleibow) [reply src/SingleProject/Resizetizer/src/AndroidAdaptiveIconGenerator.cs]: Nice!

**Comment by mattleibow** [comment]
@jonathanpeppers  researching and trying to repro those icon crashes, I believe that this PR might solve it.

For some reason, icons were crashing those devices, but we could not repro. It may also have been a bad vector in our first passes of this code. The logs showed a crash in the vector part of the stack.

Regardless, our new code uses raster images and not only is supported on older platforms, but we bypass any potential issues with custom launchers not being able to render vectors.

**Review by jonathanpeppers — APPROVED** [review]

**Pull request merged**